portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,34 +24,51 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.ws.rs.core.MediaType;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   private Gson gson = new Gson();
   private DatastoreInterface datastoreInterface = new DatastoreInterface();
+  private UserService userService = UserServiceFactory.getUserService();
+
+  private class DataPackage {
+      private boolean isUserLoggedIn;
+      private List<Comment> comments;
+      DataPackage(){
+          this.isUserLoggedIn = DataServlet.this.userService.isUserLoggedIn();
+          this.comments = DataServlet.this.datastoreInterface.getComments();
+      }
+
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(MediaType.APPLICATION_JSON);
-    String json = commentsToJson();
+    String json = dataToJson();
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
       //TODO: use GSON deserialization
-      String name = request.getParameter("name");
-      String content = request.getParameter("content");
-      Comment comment = new Comment(name, content);
-      datastoreInterface.addComment(comment);
-      response.sendRedirect("/index.html#comment-section");
+      if(userService.isUserLoggedIn()){
+        String name = request.getParameter("name");
+        String content = request.getParameter("content");
+        Comment comment = new Comment(name, content);
+        datastoreInterface.addComment(comment);
+        response.sendRedirect("/index.html#comment-section");
+      }else{
+        response.sendRedirect("/login");
+      }
   }
 
-  private String commentsToJson(){
-    List<Comment> comments = datastoreInterface.getComments();
-    String json = gson.toJson(comments);
+  private String dataToJson(){
```

Comment:
Rename this to something like `getJsonData()`.

https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names